packages/destination-actions/src/destinations/blackbaud-raisers-edge-nxt/createGift/index.ts

```diff
@@ -0,0 +1,322 @@
+        }
+      }
+    }
+  })
+
+  let constituentId = payload.constituent_id
+  if (Object.keys(constituentPayload).length > 0) {
+    // create or update constituent
+    const createOrUpdateIndividualConstituentResponse = await performCreateOrUpdateIndividualConstituent(request, {
+      payload: constituentPayload
+    })
+    constituentId = createOrUpdateIndividualConstituentResponse.id
+  } else if (constituentId === undefined) {
+    // constituent_id is required to create a gift
+    // no constituent_id, throw an error
+    throw new IntegrationError(`Missing constituent_id value`, 'MISSING_REQUIRED_FIELD', 400)
+  }
+
+  const blackbaudSkyApiClient: BlackbaudSkyApi = new BlackbaudSkyApi(request)
+
+  // data for gift call
+  const giftData: Gift = {
+    constituent_id: constituentId,
+    // hardcode is_manual
+    is_manual: true
+  }
+  const simpleGiftFields = [
+    'amount',
+    'date',
+    'gift_status',
+    'is_anonymous',
+    'lookup_id',
+    'post_date',
+    'post_status',
+    'subtype',
+    'type'
+  ]
+  simpleGiftFields.forEach((key: string) => {
+    if (payload[key] !== undefined) {
+      giftData[key] = payload[key]
+    }
+  })
+
+  // default type
+  if (!giftData.type) {
+    giftData.type = 'Donation'
+  }
+
+  // create acknowledgements array
+  if (payload.acknowledgement) {
+    const acknowledgementData: GiftAcknowledgement = {
+      status: payload.acknowledgement.status || 'NEEDSACKNOWLEDGEMENT'
+    }
+    if (
+      acknowledgementData.status !== 'NEEDSACKNOWLEDGEMENT' &&
+      acknowledgementData.status !== 'DONOTACKNOWLEDGE' &&
+      payload.acknowledgement.date
+    ) {
+      acknowledgementData.date = payload.acknowledgement.date
+    }
+    giftData.acknowledgements = [acknowledgementData]
+  }
+
+  // create gift splits array
+  giftData.gift_splits = [
+    {
+      amount: giftData.amount,
+      fund_id: payload.fund_id
+    }
+  ]
+
+  // create payments array
+  giftData.payments = [
+    {
+      payment_method: payload.payment_method
+    }
+  ]
+
+  // fields for check gifts
+  if (giftData.payments[0].payment_method === 'PersonalCheck') {
+    giftData.payments[0].check_number = payload.check_number
+    if (payload.check_date) {
+      const checkDateFuzzyDate = dateStringToFuzzyDate(payload.check_date)
+      if (checkDateFuzzyDate) {
+        giftData.payments[0].check_date = checkDateFuzzyDate
+      }
+    }
+  }
+
+  // default post status and date
+  if (!giftData.post_status) {
+    giftData.post_status = 'NotPosted'
+  }
+  if ((giftData.post_status === 'NotPosted' || giftData.post_status === 'Posted') && !giftData.post_date) {
+    giftData.post_date = payload.date
+  }
+
+  // create receipts array
+  if (payload.receipt) {
+    const receiptData: GiftReceipt = {
+      status: payload.receipt.status || 'NEEDSRECEIPT'
+    }
+    if (receiptData.status !== 'NEEDSRECEIPT' && receiptData.status !== 'DONOTRECEIPT' && payload.receipt.date) {
```

Comment:
Would it make more sense to just match on `RECEIPTED`?

Reply:
Done.